test/search_test.dart

```diff
@@ -127,6 +128,19 @@ void main() {
         var result = await index.search('prince', filter: 'tag = Tale');
         expect(result.hits, hasLength(1));
       });
+      
+      test('filter parameter is null', () async {
+        var index = await createBooksIndex();
+        var response = await index
+            .updateSettings(IndexSettings(
+              filterableAttributes: ['tag'],
+            ))
+            .waitFor(client: client);
+        expect(response.status, 'succeeded');
```

Comment:
Since you're using the `waitFor`, you'll receive a success status or the test it will fail anyway
```suggestion

```